File: erlang/rna-transcription/rna_transcription.erl
-module(rna_transcription).
-export([to_rna/1]).

to_rna(Strand) -> lists:map(fun to_rna_base/1, Strand).

to_rna_base($G) -> $C;
to_rna_base($C) -> $G;
to_rna_base($T) -> $A;
to_rna_base($A) -> $U.

File: exercism/ocaml/rna-transcription/rna_transcription.ml
open Core

type dna = [`A | `C | `G | `T]

type rna = [`A | `C | `G | `U]

let to_rna dna =
  let to_rna_base = function `G -> `C | `C -> `G | `T -> `A | `A -> `U in
  List.map ~f:to_rna_base dna
